### DISCUSSION TITLE
Iniciando a carreira como engenheiro de software

### DISCUSSION THREAD

**Discussion by emerson-d-lopes** [frontendbr/forum#1330] [Discussão]
Boa noite pessoal! 

Bem, eu começei a aprender programação no fim de 2016 e hoje me considero um desenvolvedor junior, embora ainda duvide das minhas habilidades.

Eu sei Javascript, CSS/Sass, HTML5, React, Node, Mongo, SQL e algumas outras coisinhas. O que importa é que eu sei construir coisas, talvez não da melhor maneira ou mais rapidamente mas eu sei.

No meio do ano eu comecei o curso de Sitemas de Infomação em uma universidade publica, nesse mesmo periodo eu consegui meus primeiros jobs como freelancer. 

Eu trabalhei como freela pra uma empresa durante uns 6 meses, e terminei 2 projetos, no último eu não tava trabalhando tantas horas por semana e eles resolveram terminar o contrato. Na real eu não tava trabalhando tanto porque me sentia cansado e muitas das vezes não entendia o codigo (que outro desenvolvedor escreveu) e tbm estava no fim do semestre na universidade.

Eu peguei algums outros projetos no upwork tbm, mas bem mais simples. Sempre que eu tento aplicar para novos projetos eu consigo algumas entrevistas mas não passa disso (as vezes por falta de experiencia ou por não poder trabalhar 30+ por semana). O que eu tiro disso é que eu sou bom o suficiente para conseguir chegar nessas entrevistas.

Então, por que eu to falei tudo isso? 

Eu não necessariamente preciso trabalhar agora, mas tenho o feeling que seria legal para minha carreira. Por outro lado tem a universidade, em que materiais relacionadas a programação, por enquanto, são tranquilas para mim, só levo pau em matematica e no mundo js tem sempre um monte de coisas novas para aprender. 

O que vcs me aconselham a fazer? Me dedicar mais aos estudos, tanto na universidade quanto sozinho, fazendo projetos pessoais e tals para aprender mais a fundo sobre programação ou meter as caras no mercado e construir projetos reais (se eu conseguir algum) ?

Ps: Como eu disse eu me sinto confiante com js mas ainda bate o sentimento de que eu não sou tão bom, eu senti isso especialmente quando eu tava trabalhando, mas tbm sentia que eu era foda quando conseguia fazer algo.

E aí? Estudar? Trabalhar? Projetos pessoais ? Fazer os 3 ao mesmo tempo?

**Comment by isacmoura** [comment]
Você não precisa aprender todos os frameworks JS, aprende alguns mais usados. Eu também me sentia assim, sem confiança pra fazer algo funcional do zero, mas você só vai saber se tentar. Obviamente vai demorar um pouco mais pra fazer algo que você nunca viu, mas velho, é pesquisar na internet e implementar.

Se você não se sente confiante no momento, eu recomendo focar um pouco mais na faculdade e em projetos pessoais (lembre-se também de contribuir em projetos open). Tenta balancear as cadeiras da faculdade, colocando 1 difícil e algumas fáceis etc. Comigo isso vem funcionando.

**Comment by simaodeveloper** [comment]
**Não sobrecarregue sua mente**

@lupuselit tudo bom? vamos por partes, vi que tem um conhecimento amplo, acho isso muito bom, mas precisa ir com calma, está fazendo faculdade, pegando alguns freelas, acho isso muito massa, eu mesmo trabalho full-time (44hrs/semana) e pego alguns freelas para testar novos conhecimentos adquiridos estudando sozinho.

_Mas tem que ir com calma, não sobrecarregue a sua mente, o estresse e a ansiedade vão atrapalhar seus sonhos, essas doenças refletem diretamente no nosso corpo._

Quando pegar um freela, pegue **jobs pequenos** e vai focando em **constuir seu portfólio**, caso pegar um código complexo, se depois de 30 min, mesmo se não tiver quebrado em pequenas partes e ir resolvendo uma a uma e você ainda não encontrar a solução, você tem a obrigação de **jogar sua dúvida aqui no grupo, eu tenho certeza que há pessoas assim como eu dispostas a te ajudar, mesmo que você acha que é uma dúvida boba, para mim só existe dúvida e vamos te ajudar a resolver e entender o porque está com dificuldade.**

logo logo, caso quiser, poderá aplicar para alguma vaga e terá que conciliar a faculdade e trabalho, mas relaxa é assim mesmo.

A vida é assim temos que conciliar, trabalho, faculdade, amigos, familia, você vai se ajustar e dar tempo para cada coisa, tamo junto nessa vida, somos todos fudidos buscando uma vida melhor!

Abraços qualquer coisa me procura no telegram @danielsimao

**Comment by camillodev** [comment]
Se você não consegue trabalhar mais de 30hs por semana e ta fazendo faculdade, foca nas vagas de estagio e dependendo do periodo rola trainee também. 
Além disso você citou que as vezes não se acha bom o suficiente. Da uma olhada nesse [post aqui](https://willianjusten.com.br/sou-bom-o-suficiente/) do @willianjusten . 
Ele também fez um [post ](https://willianjusten.com.br/coisas-que-eu-nao-sei-em-ti/) recentemente falando das coisas que ele não sabe, e acho que é um otimo complemento. 

Sobre conseguir chegar até as entrevistas, significa que seu curriculo ta bom, agora foca em se preparar para as entrevistas. Qual parte que você não passa, na conversa com RH, nos testes tecnicos? E foca mais nisso. Tem um curso gratuito da Udacity de [preparação para entrevistas de Front End](https://www.udacity.com/course/front-end-interview-prep--ud250)
